common/tests/test_tls.py

```diff
@@ -0,0 +1,26 @@
+import contextlib
+import os
+
+import pytest
+import requests
+
+from determined_common.api import request
+
+TRUSTED_DOMAIN = "https://example.com"
+UNTRUSTED_DOMAIN = "https://untrusted-root.badssl.com"
+UNTRUSTED_CERT_FILE = os.path.join(os.path.dirname(__file__), "untrusted-root.badssl.com.crt")
+
+
+def test_custom_tls_certs() -> None:
+    for bundle, raises in [
+        (False, False),
+        (True, True),
+        (UNTRUSTED_CERT_FILE, False),
+        (None, True),
+    ]:
+        request.set_master_cert_bundle(bundle)  # type: ignore
+
+        request.get(TRUSTED_DOMAIN, "", authenticated=False)
+        ctx = pytest.raises(requests.exceptions.SSLError) if raises else contextlib.ExitStack()
```

Comment:
mabye use `contextlib.nullcontext` instead of `ExitStack`?  Or use `contextlib.ExitStack` always and optionally call `stack.enter_context()` on `pytest.raises`?

Reply:
`nullcontext` is new in 3.7, so no good here. But I like the idea of actually using `ExitStack` for what it's for.